ui/src/app/pages/admin/admin-user-common.tsx

```diff
@@ -0,0 +1,325 @@
+    color: colors.accent,
+    borderRadius: '18px',
+    transform: 'rotate(270deg)'
+  },
+});
+
+export const adminGetProfile = async(usernameWithDomain: string): Promise<Profile> => {
+  return userAdminApi().getUserByUsername(usernameWithDomain);
+}
+
+export const getPublicInstitutionDetails = async(): Promise<PublicInstitutionDetails[]> => {
+  const institutionsResponse = await institutionApi().getPublicInstitutionDetails();
+  const institutions = institutionsResponse?.institutions;
+  return fp.sortBy('displayName', institutions);
+}
+
+/**
+ * Present an ordered list of dollar options with the following values:
+ * $300 to $1000, in $100 increments
+ * $1000 to $10,000, in $500 increments
+ * Plus the user's current quota value, if it's not already one of these
+ */
+export const getFreeCreditLimitOptions = (freeTierDollarQuota?: number) => {
+  const START1 = 300;
+  const END1 = 1000;
+  const END2 = 10000;
+
+  // gotcha: argument order for rangeStep is (step, start, end)
+  // IntelliJ incorrectly believes the order is (start, end, step)
+  const below1000 = fp.rangeStep(100, START1, END1+1);
+  const over1000 = fp.rangeStep(500, END1, END2+1);
+
+  const defaultsPlusMaybeOverride = new Set([...below1000, ...over1000, freeTierDollarQuota ?? START1]);
+
+  // gotcha: JS sorts numbers lexicographically by default
+  const numericallySorted = Array.from(defaultsPlusMaybeOverride).sort((a, b) => a - b);
+
+  return fp.map((limit) => ({label: formatFreeCreditsUSD(limit), value: limit}), numericallySorted);
+}
+
+export const getFreeCreditUsage = (profile: Profile): string => {
+  const {freeTierDollarQuota, freeTierUsage} = profile;
+  return `${formatFreeCreditsUSD(freeTierUsage)} used of ${formatFreeCreditsUSD(freeTierDollarQuota)} limit`;
+}
+
+// returns the updated profile value only if it has changed
+export const getUpdatedProfileValue = (oldProfile: Profile, updatedProfile: Profile, attribute: string) => {
+  const oldValue = fp.get([attribute], oldProfile);
+  const updatedValue = fp.get([attribute], updatedProfile);
+  if (!fp.isEqual(oldValue, updatedValue)) {
+    return updatedValue;
+  } else {
+    return null;
+  }
+}
+
+export const enableSave = (oldProfile: Profile, updatedProfile: Profile, errors): boolean =>
+  !errors && !fp.isEqual(oldProfile, updatedProfile);
+
+export const updateAccountProperties = async(oldProfile: Profile, updatedProfile: Profile): Promise<Profile> => {
+  const {username} = updatedProfile;
+  const request: AccountPropertyUpdate = {
+    username,
+    freeCreditsLimit: getUpdatedProfileValue(oldProfile, updatedProfile, 'freeTierDollarQuota'),
+    contactEmail: getUpdatedProfileValue(oldProfile, updatedProfile, 'contactEmail'),
+    affiliation: getUpdatedProfileValue(oldProfile, updatedProfile, 'verifiedInstitutionalAffiliation'),
+    accessBypassRequests: [],  // coming soon: RW-4958
+  };
+
+  return userAdminApi().updateAccountProperties(request);
+}
+
+export const DropdownWithLabel =
```

Comment:
Not part of this PR, but we should check if we can use this in other files/pages and move to a utils file

Reply:
actually we may want to phase out Dropdown: https://precisionmedicineinitiative.atlassian.net/browse/RW-6023

Reply:
cool